src/ert/config/_read_summary.py

```diff
@@ -0,0 +1,445 @@
+                try:
+                    date = datetime(
+                        day=day,
+                        month=month,
+                        year=year,
+                        hour=hour,
+                        minute=minute,
+                        second=microsecond // 10**6,
+                        microsecond=microsecond % 10**6,
+                    )
+                except Exception as err:
+                    raise ValueError(
+                        f"SMSPEC {spec} contains invalid STARTDAT: {err}"
+                    ) from err
+    keywords = arrays["KEYWORDS"]
+    wgnames = arrays["WGNAMES "]
+    nums = arrays["NUMS    "]
+    numlx = arrays["NUMLX   "]
+    numly = arrays["NUMLY   "]
+    numlz = arrays["NUMLZ   "]
+    lgr_names = arrays["LGRS    "]
+
+    if date is None:
+        raise ValueError(f"Keyword startdat missing in {spec}")
+    if keywords is None:
+        raise ValueError(f"Keywords missing in {spec}")
+    if n is None:
+        n = len(keywords)
+
+    indices: List[int] = []
+    keys: List[str] = []
+    date_index = None
+
+    def optional_get(arr: Optional[npt.NDArray[Any]], idx: int) -> Any:
+        if arr is None:
+            return None
+        if len(arr) <= idx:
+            return None
+        return arr[idx]
+
+    for i in range(n):
+        keyword = _key2str(keywords[i])
+        if keyword == "TIME":
+            date_index = i
+
+        name = optional_get(wgnames, i)
+        if name is not None:
+            name = _key2str(name)
+        num = optional_get(nums, i)
+        lgr_name = optional_get(lgr_names, i)
+        if lgr_name is not None:
+            lgr_name = _key2str(lgr_name)
+        li = optional_get(numlx, i)
+        lj = optional_get(numly, i)
+        lk = optional_get(numlz, i)
+
+        key = make_summary_key(keyword, num, name, nx, ny, lgr_name, li, lj, lk)
+        if key is not None and _should_load_summary_key(key, fetch_keys):
+            if key in keys:
+                # only keep the index of the last occurrence of a key
+                # this is done for backwards compatability
+                # and to have unique keys
+                indices[keys.index(key)] = i
+            else:
+                indices.append(i)
+                keys.append(key)
+
+    keys_array = np.array(keys)
+    rearranged = keys_array.argsort()
+    keys_array = keys_array[rearranged]
+
+    indices_array = np.array(indices)[rearranged]
+
+    units = arrays["UNITS   "]
+    if units is None:
+        raise ValueError(f"Keyword units missing in {spec}")
+    if date_index is None:
+        raise ValueError(f"KEYWORDS did not contain TIME in {spec}")
+    if date_index >= len(units):
+        raise ValueError(f"Unit missing for TIME in {spec}")
+
+    unit_key = _key2str(units[date_index])
+    try:
+        date_unit = DateUnit[unit_key]
+    except KeyError:
+        raise ValueError(f"Unknown date unit in {spec}: {unit_key}") from None
```

Comment:
Why dont you want the context here?

Reply:
I think the keyerror is just misleading compared to the ValueError raised.